src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/RSACryptoServiceProvider.Unix.cs

```diff
@@ -0,0 +1,241 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.Cryptography;
+using Internal.NativeCrypto;
+using System.IO;
+
+namespace System.Security.Cryptography
+{
+    public sealed partial class RSACryptoServiceProvider : RSA, ICspAsymmetricAlgorithm
+    {
+        private readonly RSA _impl;
+        private bool _publicOnly;
+
+        public RSACryptoServiceProvider()
+            : this(0, true) { }
+
+        public RSACryptoServiceProvider(int dwKeySize)
+            : this(dwKeySize, false) { }
+
+        public RSACryptoServiceProvider(int dwKeySize, CspParameters parameters)
+            : this(dwKeySize, false) { }
```

Comment:
OK. This was actually a question I was going to ping you on, as whether we should ignore or throw. I made the DSA throw in this case.